guides/03_additional-features/02_streaming-outputs.md

```diff
@@ -18,3 +18,25 @@ $demo_fake_diffusion
 Note that we've added a `time.sleep(1)` in the iterator to create an artificial pause between steps so that you are able to observe the steps of the iterator (in a real image generation model, this probably wouldn't be necessary).
 
 Similarly, Gradio can handle streaming inputs, e.g. an image generation model that reruns every time a user types a letter in a textbox. This is covered in more details in our guide on building [reactive Interfaces](/guides/reactive-interfaces). 
+
+## Streaming Media
+
+Gradio can stream audio and video directly from your generator function.
+This lets your user hear your audio or see your video nearly as soon as it's `yielded` by your function.
+All you have to do is 
+
+1. Set `streaming=True` in your `gr.Audio` or `gr.Video` output component.
+1. Write a python generator that yields the next "chunk" of audio or video.
+
+For audio, the next "chunk" can be either an `.mp3` or `.wav` file or a `bytes` sequence of audio.
+For video, the next "chunk" has to be either `.mp4` file or a file with `h.264` codec with a `.ts` extension.
+
+Let's look at some examples
+
+### Streaming Audio
```

Comment:
Nice addition to the docs. This is such an important use case imo that you could have a dedicated guide for TTS showcasing `transformers`, analogous to this one: https://www.gradio.app/guides/real-time-speech-recognition

Reply:
Also, since this code example is quite complicated, you might want to start with a simple example just to showcase usage, like this:

```py
import gradio as gr
from time import sleep

def keep_repeating(audio_file):
    for i in range(10):
        sleep(0.5)
        yield audio_file

gr.Interface(keep_repeating,
             gr.Audio(sources=["microphone"], type="filepath"),
             gr.Audio(streaming=True, autoplay=True)
).launch()
```

Reply:
Yes good call on switching the demo code!

Reply:
Will add guide in a separate PR